app/channel.go

```diff
@@ -2724,11 +2724,15 @@ func (a *App) markChannelAsUnreadFromPostCRTUnsupported(postID string, userID st
 		}
 		// If threadmembership already exists but user had previously unfollowed the thread, then follow the thread again.
 		threadMembership.Following = true
-		threadMembership.LastViewed = post.UpdateAt - 1
-		threadMembership.UnreadMentions, err = a.countThreadMentions(user, rootPost, channel.TeamId, post.UpdateAt-1)
+		threadMembership.LastViewed = post.CreateAt - 1
+		threadMembership.UnreadMentions, err = a.countThreadMentions(user, rootPost, channel.TeamId, post.CreateAt-1)
```

Comment:
Related fix:
These should be CreateAt, not UpdateAt. The subsequent methods also filter posts / count messages by CreateAt, so using UpdateAt will lead to less than the expected number of Mentions/Unreads under some circumstances.

Needs tests.